conf/conf_test.go

```diff
@@ -67,3 +84,518 @@ func TestGetYamlFileFromFile(t *testing.T) {
+	for _, s := range n.Slack {
+		checkSlackNotify(t, s)
+	}
+	for _, d := range n.Discord {
+		checkDiscordNotify(t, d)
+	}
+	for _, tg := range n.Telegram {
+		checkTelegramNotify(t, tg)
+	}
+	for _, e := range n.Email {
+		checkEmailNotify(t, e)
+	}
+}
+
+func checkSlackNotify(t *testing.T, n slack.NotifyConfig) {
+	switch n.NotifyName {
+	case "Slack":
+		assert.Equal(t, n.WebhookURL, "https://hooks.slack.com/services/xxx")
+		assert.Equal(t, n.Channels(), []string{"slack", "general"})
+		assert.Equal(t, n.Dry, false)
+	default:
+		t.Errorf("unexpected notify name %s", n.NotifyName)
+	}
+}
+
+func checkDiscordNotify(t *testing.T, n discord.NotifyConfig) {
+	switch n.NotifyName {
+	case "Discord":
+		assert.Equal(t, n.WebhookURL, "https://discord.com/api/webhooks/xxx")
+		assert.Equal(t, n.Avatar, "https://img.icons8.com/ios/72/appointment-reminders--v1.png")
+		assert.Equal(t, n.Thumbnail, "https://freeiconshop.com/wp-content/uploads/edd/notification-flat.png")
+		assert.Equal(t, n.Dry, true)
+		assert.Equal(t, n.Retry.Times, 3)
+		assert.Equal(t, n.Retry.Interval, 10*time.Second)
+	case "Test":
+		assert.Equal(t, n.WebhookURL, "https://discord.com/api/webhooks/xxx")
+		assert.Equal(t, n.Avatar, "https://img.icons8.com/ios/72/appointment-reminders--v1.png")
+		assert.Equal(t, n.Thumbnail, "https://freeiconshop.com/wp-content/uploads/edd/notification-flat.png")
+		assert.Equal(t, n.Dry, false)
+		assert.Equal(t, n.Retry.Times, 3)
+		assert.Equal(t, n.Retry.Interval, 10*time.Second)
+		assert.Equal(t, n.Channels(), []string{"general"})
+
+	default:
+		t.Errorf("unexpected notify name %s", n.NotifyName)
+	}
+}
+
+func checkTelegramNotify(t *testing.T, n telegram.NotifyConfig) {
+	switch n.NotifyName {
+	case "Dev Channel":
+		assert.Equal(t, n.Token, "123456:xxxxx")
+		assert.Equal(t, n.ChatID, "-1001343458903")
+		assert.Equal(t, n.Channels(), []string{"telegram#Dev", "test"})
+		assert.Equal(t, n.Dry, false)
+	case "Ops Group":
+		assert.Equal(t, n.Token, "123456:yyyyyy")
+		assert.Equal(t, n.ChatID, "-12310934")
+		assert.Equal(t, n.Dry, true)
+		assert.Equal(t, n.Retry.Times, 3)
+		assert.Equal(t, n.Retry.Interval, 8*time.Second)
+	default:
+		t.Errorf("unexpected notify name %s", n.NotifyName)
+	}
+}
+
+func checkEmailNotify(t *testing.T, n email.NotifyConfig) {
+	switch n.NotifyName {
+	case "email":
+		assert.Equal(t, n.Server, "smtp.email.com:465")
+		assert.Equal(t, n.User, "noreply@megaease.com")
+		assert.Equal(t, n.Pass, "xxx")
+		assert.Equal(t, n.From, "send@email.com")
+		assert.Equal(t, n.To, "test@email.com")
+		assert.Equal(t, n.Dry, true)
+	default:
+		t.Errorf("unexpected notify name %s", n.NotifyName)
+	}
+}
+
+const confSettings = `
+settings:
+  name: "EaseProbeBot"
+  icon: https://upload.wikimedia.org/wikipedia/commons/2/2d/Etcher-icon.png
+  http:
+    ip: 127.0.0.1
+    port: 8181
+    refresh: 5s
+    log:
+      file: /tmp/log/easeprobe.access.log
+      self_rotate: true
+  sla:
+    schedule: "weekly"
```

Comment:
The test coverage of the file `conf.go`  is lower than 76.2%, we could promote the rate via adding schedule value and introducing the `https://github.com/bouk/monkey` to mock the os.GetEnv , os.MkDir.

```golang
schedules := []string{"hourly", "daily","weekly", ...}
const = confSettings = `
...
   sla:
    schedule: "weekly"
    time: "23:59"
    debug: true
... 
`    
func TestConfig(t *testing.T) {
    for i, schedule := range schedules { 
	file := "./config.yaml"
	err := writeConfig(file, scheule) //         setting := fmt.Sprintf(confSettings, schedule)
	assert.Nil(t, err)
...      
    }
}

```